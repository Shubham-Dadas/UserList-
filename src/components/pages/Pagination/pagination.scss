.pagination-container {
    display: flex;
    justify-content: center;
    align-items: center;
    gap: 0.375rem;
    margin-top: 0.625rem;
}

.pagination-btn {
    background-color: #fff;
    color: #6A1E55;
    border: 0.125rem solid #6A1E55;
    padding: 0.375rem 0.75rem;
    font-size: 1rem;
    font-weight: bold;
    cursor: pointer;
    border-radius: 0.3125rem;
    transition: 0.3s ease;
}

.pagination-btn:hover {
    background-color: #6A1E55;
    color: white;
}

.pagination-btn.active {
    background-color: #6A1E55;
    color: white;
}

.pagination-btn:disabled {
    background-color: #ddd;
    border: 0.125rem solid #bbb;
    color: #999;
    cursor: not-allowed;
}

.pagination-dots {
    padding: 0.375rem 0.75rem;
    font-size: 1rem;
    font-weight: bold;
    color: #6A1E55;
}

.prev,
.next {
    background-color: #6A1E55;
    color: white;
    border: none;
}

.prev:hover,
.next:hover {
    background-color: #7a2955;
}