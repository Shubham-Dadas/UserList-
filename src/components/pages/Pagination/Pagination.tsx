import React from "react";
import "./pagination.scss";

interface PaginationProps {
  handlePageClick: (data: { selected: number }) => void;
  pageCount: number;
  currentPage: number;
}

const Pagination: React.FC<PaginationProps> = ({
  handlePageClick,
  pageCount,
  currentPage,
}) => {
  const getPageNumbers = () => {
    let pages: (number | string)[] = [];

    if (pageCount <= 4) {
      pages = Array.from({ length: pageCount }, (_, i) => i);
    } else {
     
      pages.push(0);

      if (currentPage > 2) pages.push("...");

      let start = Math.max(1, currentPage - 1);
      let end = Math.min(pageCount - 2, currentPage + 1);

      for (let i = start; i <= end; i++) {
        pages.push(i);
      }

      if (currentPage < pageCount - 3) pages.push("...");
        pages.push(pageCount - 1);
      }

    return pages;
  };

  return (
    <div className="pagination-container">
      <button
        onClick={() =>
          handlePageClick({ selected: Math.max(0, currentPage - 1) })
        }
        disabled={currentPage === 0}
        className="pagination-btn prev"
      >
        ◄
      </button>

      {getPageNumbers().map((page, index) =>
        page === "..." ? (
          <span key={index} className="pagination-dots">
            ...
          </span>
        ) : (
          <button
            key={index}
            onClick={() => handlePageClick({ selected: page as number })}
            className={`pagination-btn ${currentPage === page ? "active" : ""}`}
          >
            {Number(page) + 1}
          </button>
        )
      )}

      <button
        onClick={() =>
          handlePageClick({
            selected: Math.min(pageCount - 1, currentPage + 1),
          })
        }
        disabled={currentPage === pageCount - 1}
        className="pagination-btn next"
      >
        ►
      </button>
    </div>
  );
};

export default Pagination;
