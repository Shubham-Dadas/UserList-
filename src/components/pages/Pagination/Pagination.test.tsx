import Pagination from "./Pagination";
import { mount, shallow } from "enzyme";
import React from "react";
import toJson from "enzyme-to-json";

const mockHandlePageClick = jest.fn();

const props = {
  handlePageClick: mockHandlePageClick,
  pageCount: 5,
  currentPage: 2
};

it("check mounting of pagination component", () => {
    const component = shallow(<Pagination {...props} />);
    // @ts-ignore
    expect(toJson(component)).toMatchSnapshot();
})

it("check disable of prev button for first page", () => {
  const component = shallow(<Pagination {...props} currentPage={0}/>);
  // @ts-ignore
  expect(component.find(".prev").prop("disabled")).toBe(true)
})

it("check disable of next button for last page", () => {
  const component = shallow(<Pagination {...props} currentPage={4} />);
  // @ts-ignore
  expect(component.find(".next").prop("disabled")).toBe(true);
});

it("finding button in component", () => {
    const component = shallow(<Pagination {...props} currentPage={2} />)
    expect(component.find("button").exists()).toBe(true)
})

it("check calling of function", () => { 
    const component = mount(<Pagination {...props} />);
    component.find(".prev").simulate("click");
    expect(mockHandlePageClick).toHaveBeenCalledWith({ selected: 1 });
})
