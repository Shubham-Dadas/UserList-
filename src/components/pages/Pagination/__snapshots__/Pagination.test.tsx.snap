// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check Pagination mounting of pagination component 1`] = `
<div
  className="pagination-container"
>
  <button
    className="pagination-btn prev"
    disabled={false}
    onClick={[Function]}
  >
    ◄
  </button>
  <button
    className="pagination-btn "
    key="0"
    onClick={[Function]}
  >
    1
  </button>
  <button
    className="pagination-btn "
    key="1"
    onClick={[Function]}
  >
    2
  </button>
  <button
    className="pagination-btn active"
    key="2"
    onClick={[Function]}
  >
    3
  </button>
  <button
    className="pagination-btn "
    key="3"
    onClick={[Function]}
  >
    4
  </button>
  <button
    className="pagination-btn "
    key="4"
    onClick={[Function]}
  >
    5
  </button>
  <button
    className="pagination-btn next"
    disabled={false}
    onClick={[Function]}
  >
    ►
  </button>
</div>
`;

exports[`check mounting of pagination component 1`] = `
<div
  className="pagination-container"
>
  <button
    className="pagination-btn prev"
    disabled={false}
    onClick={[Function]}
  >
    ◄
  </button>
  <button
    className="pagination-btn "
    key="0"
    onClick={[Function]}
  >
    1
  </button>
  <button
    className="pagination-btn "
    key="1"
    onClick={[Function]}
  >
    2
  </button>
  <button
    className="pagination-btn active"
    key="2"
    onClick={[Function]}
  >
    3
  </button>
  <button
    className="pagination-btn "
    key="3"
    onClick={[Function]}
  >
    4
  </button>
  <button
    className="pagination-btn "
    key="4"
    onClick={[Function]}
  >
    5
  </button>
  <button
    className="pagination-btn next"
    disabled={false}
    onClick={[Function]}
  >
    ►
  </button>
</div>
`;
